be/src/exec/parquet_reader.cpp

```diff
@@ -166,6 +176,57 @@ Status ParquetReaderWrap::init_parquet_reader(const std::vector<SlotDescriptor*>
     }
 }
 
+Status ParquetReaderWrap::get_schema(std::vector<SlotDescriptor>* schema) {
+    RETURN_IF_ERROR(_init_parquet_reader());
+
+    auto file_schema = _file_metadata->schema();
+    for (int i = 0; i < file_schema->num_columns(); ++i) {
+        auto column = file_schema->Column(i);
+        auto name = column->name();
+        auto physical_type = column->physical_type();
+        auto logical_type = column->logical_type();
+
+        // See detail in https://github.com/apache/parquet-format/blob/master/LogicalTypes.md.
+        TypeDescriptor tp;
+        switch (physical_type) {
+        case parquet::Type::BOOLEAN:
+            tp = TypeDescriptor(TYPE_BOOLEAN);
+            break;
+        case parquet::Type::FLOAT:
+            tp = TypeDescriptor(TYPE_FLOAT);
+            break;
+        case parquet::Type::DOUBLE:
+            tp = TypeDescriptor(TYPE_DOUBLE);
+            break;
+        case parquet::Type::INT32:
+            if (logical_type->is_date()) {
+                tp = TypeDescriptor(TYPE_DATE);
+            } else {
+                tp = TypeDescriptor(TYPE_INT);
+            }
+            break;
+        case parquet::Type::INT64:
+            if (logical_type->is_timestamp()) {
+                tp = TypeDescriptor(TYPE_DATETIME);
+            } else {
+                tp = TypeDescriptor(TYPE_BIGINT);
+            }
+            break;
+        case parquet::Type::INT96:
+            tp = TypeDescriptor(TYPE_DATETIME);
+            break;
+        case parquet::Type::BYTE_ARRAY:
```

Comment:
some logic types may also annotate `BYTE_ARRAY`

Reply:
Updated. The BYTE_ARRAY is parsed as VARCHAR or DECIMAL according to the logical type.